applauncher/003-turbo-list.qml

```diff
@@ -0,0 +1,205 @@
+ *               2011 Tom Swindell <t.swindell@rubyx.co.uk>
+ * All rights reserved.
+ *
+ * You may use this file under the terms of BSD license as follows:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the author nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import QtQuick 2.15
+import QtGraphicalEffects 1.12
+import org.asteroid.controls 1.0
+import org.asteroid.utils 1.0
+
+ListView {
+    id: appsView
+    anchors.fill: parent
+    preferredHighlightBegin: appsView.height/2 - appsView.height/12
+    preferredHighlightEnd: appsView.height/2 + appsView.height/12
+    highlightRangeMode: ListView.StrictlyEnforceRange
+
+    property int currentPos: 0
+    // Scaling for when the launcher is shown as full screen (1.0) or in the settings app (0.5).
+    property var viewScale: (appsView.width > appsView.height ? appsView.height : appsView.width) / Dims.l(100)
+
+    onCurrentPosChanged: {
+        rightIndicator.animate()
+        leftIndicator.animate()
+        topIndicator.animate()
+        bottomIndicator.animate()
+    }
+
+    Connections {
+        target: grid
+        function onNormalizedVerOffsetChanged() {
+            // Override wallpaper darkening, so that the default flatmesh color is used.
+            if (!displayAmbient) {
+                wallpaperAnimation.duration = 0
+
+                wallpaper.anchors.verticalCenterOffset = height*grid.normalizedVerOffset*(-0.05)
+                wallpaperDarkener.opacity = Math.abs(grid.normalizedVerOffset)*0.4
```

Comment:
Ugh, this is very hacky. I would rather have the main asteroid-launcher codebase offer an easy boolean to disable this rather than introducing this dependency. This breaks abstraction layers and introduces a lot of complexity (likt the !displayAmbient condition)

Reply:
Agreed on the hacky nature of this.
I've adjusted this so that a global boolean is used `launcherColorOverride`. It reduces the overall code size and duplication too :smile: